lib/puppet_x/puppetlabs/property/region.rb

```diff
@@ -0,0 +1,15 @@
+module PuppetX
+  module Property
+    class AwsRegion < Puppet::Property
+      validate do |value|
```

Comment:
I suspect the failure cases can be cleaned up here a bit.  If we get a value, is it a string, else fail.  If its a string, match a regex to ensure its a valid region.